src/Compilers/CSharp/Portable/Symbols/Tuples/TupleErrorFieldSymbol.cs

```diff
@@ -124,7 +124,10 @@ public bool Equals(TupleErrorFieldSymbol other)
                 return true;
             }
 
-            return (object)other != null && _tupleFieldId == other._tupleFieldId && ContainingType == other.ContainingType;
+            return (object)other != null &&
+                _tupleElementIndex == other._tupleElementIndex &&
+                ContainingType == other.ContainingType &&
+                Name == other.Name;
```

Comment:
>Name == other.Name [](start = 16, length = 18)

Given the equality of the containing type, I think name comparison is redundant here.